cpu/cortexm_common/thread_arch.c

```diff
@@ -234,6 +238,17 @@ char *thread_arch_stack_init(thread_task_func_t task_func,
     return (char*) stk;
 }
 
+/* This function gets the number of bytes used on the ISR stack */
+int thread_arch_isr_stack_usage(void)
+{
+    int count = 0;
+    uint32_t* sp = (uint32_t *) _estack;
+    while(sp != (uint32_t *) _sstack)
+        if (*sp == STACK_CANARY_WORD)
+            count++;
+    return ((int)(512 - count));
```

Comment:
this looks over-complicated (and might be broken):
- you have to start from `_sstack`, as stacks on cortex-m platforms grow from top to bottom
- no need to use `_estack` here
- DON'T hard-code the stack size
- and please always use parenthesis when using `if/while/for...`

I would suggest something like:
```c
int count = 0;
uint32_t  *ptr = &_sstack;
if (*ptr != STACK_CANARY_WORD) {
    return -1; /* stack overflowed - something is not right */
}
while (*(ptr++) == STACK_CANARY_WORD) {
    count++;
}
return count;
```
(though I did not test this code...)